gprofiler/metadata/system_metadata.py

```diff
@@ -0,0 +1,236 @@
+    # Google's DNS servers (8.8.8.8) and getting the local IP address of the interface it connected with.
+    try:
+        private_ips = [ip for ip in socket.gethostbyname_ex(socket.gethostname())[2] if not ip.startswith("127.")]
+    except socket.error:
+        # Could happen when a network is unavailable
+        private_ips = []
+    if not private_ips:
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        try:
+            s.connect(("8.8.8.8", 53))
+            private_ips.append(s.getsockname()[0])
+        finally:
+            s.close()
+    return private_ips[0] if private_ips else "unknown"
+
+
+def get_mac_address() -> str:
+    """
+    Gets the MAC address of the first non-loopback interface.
+    """
+
+    assert sys.maxsize > 2 ** 32, "expected to run on 64-bit!"
+    SIZE_OF_STUCT_ifreq = 40  # correct for 64-bit
+
+    IFNAMSIZ = 16
+    IFF_LOOPBACK = 8
+    MAC_BYTES_LEN = 6
+    SIZE_OF_SHORT = struct.calcsize('H')
+    MAX_BYTE_COUNT = 4096
+
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_IP)
+
+    # run SIOCGIFCONF to get all interface names
+    buf = array.array('B', b'\0' * MAX_BYTE_COUNT)
+    ifconf = struct.pack('iL', MAX_BYTE_COUNT, buf.buffer_info()[0])
+    outbytes = struct.unpack('iL', fcntl.ioctl(s.fileno(), 0x8912, ifconf))[0]  # SIOCGIFCONF
+    data = buf.tobytes()[:outbytes]
+    for index in range(0, len(data), SIZE_OF_STUCT_ifreq):
+        iface = data[index : index + SIZE_OF_STUCT_ifreq]
+
+        # iface is now a struct ifreq which starts with the interface name.
+        # we can use it for further calls.
+        res = fcntl.ioctl(s.fileno(), 0x8913, iface)  # SIOCGIFFLAGS
+        ifr_flags = struct.unpack(f'{IFNAMSIZ}sH', res[: IFNAMSIZ + SIZE_OF_SHORT])[1]
+        if ifr_flags & IFF_LOOPBACK:
+            continue
+
+        # okay, not loopback, get its MAC address.
+        res = fcntl.ioctl(s.fileno(), 0x8927, iface)  # SIOCGIFHWADDR
+        address = struct.unpack(f'{IFNAMSIZ}sH{MAC_BYTES_LEN}s', res[: IFNAMSIZ + SIZE_OF_SHORT + MAC_BYTES_LEN])[2]
+        mac = struct.unpack(f'{MAC_BYTES_LEN}B', address)
+        address = ":".join(['%02X' % i for i in mac])
+        return address
+
+    return "unknown"
+
+
+@dataclass
+class SystemInfo:
+    python_version: str
+    run_mode: str
+    deployment_type: str
+    kernel_release: str
+    kernel_version: str
+    system_name: str
+    processors: int
+    memory_capacity_mb: int
+    hostname: str
+    os_name: str
+    os_release: str
+    os_codename: str
+    libc_type: str
+    libc_version: str
+    hardware_type: str
+    pid: int
+    mac_address: str
+    private_ip: str
+    spawn_uptime_ms: float
+
+
+def get_static_system_info() -> SystemInfo:
+    hostname, distribution, libc_tuple, mac_address, private_ip = _initialize_system_info()
+    clock = getattr(time, "CLOCK_BOOTTIME", time.CLOCK_MONOTONIC)
```

Comment:
Not sure that's enough. This tests if `CLOCK_BOOTTIME` exists on `time`; from CPython:
```
#ifdef CLOCK_BOOTTIME
        if (PyModule_AddIntMacro(module, CLOCK_BOOTTIME) < 0) {
            return -1;
        }
#endif
```
we're using Python built with one libc, and run it on older kernels.

Basically, this should run on CentOS 6 which is the oldest we have. This may fail and then we want `MONOTONIC`. Let's try running on that CentOS tomorrow (it's a good idea to test this PR on that old system anyway, since it makes many OS-related changes)

Reply:
```
In [4]: time.clock_gettime(time.CLOCK_BOOTTIME + 5)                                                                                                                                                        
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-4-d8bd7da901c2> in <module>
----> 1 time.clock_gettime(time.CLOCK_BOOTTIME + 5)

OSError: [Errno 22] Invalid argument
```

let's catch that error and try `MONOTONIC`?